### PR TITLE
DOCS-1429: Document Robot API remaining methods

### PR DESCRIPTION
**Pull request by sguequierre**
* `WithChannel()` in particular needs SME review.
* Add undocumented methods.
* Removed in checker `get_component` and `get_services` methods instead of documenting, bc they are an alternative method to `from_robot` that is meant to barely be called if ever.

### CODE REVIEW

File: docs/build/program/apis/robot.md
Comment (sguequierre) [inline docs/build/program/apis/robot.md:264]: @cheukt Is this accurate? How can I show the channels between two robots with Python? grabbed this from test code but doesn't look like it's making an actual channel

File: docs/build/program/apis/robot.md
Comment (sguequierre) [inline docs/build/program/apis/robot.md:264]: @cheukt Is this accurate? How can I show the channels between two robots with Python? grabbed this from test code but doesn't look like it's making an actual channel

File: docs/build/program/apis/robot.md
Comment (cheukt) [reply docs/build/program/apis/robot.md]:
```
from viam.robot.client import RobotClient
from viam.rpc.dial import DialOptions, dial

async def connect_with_channel() -> RobotClient:
    async with await dial('ADDRESS', DialOptions() as channel:
        return await RobotClient.with_channel(channel, RobotClient.Options())

robot = await connect_with_channel()
print(robot.resource_names)
await robot.close()
```
would be how you get a channel in the Python SDK

File: docs/build/program/apis/robot.md
Comment (cheukt) [inline docs/build/program/apis/robot.md]: documenting these methods here feels weird - these are specific Python library methods that are not part of the API and so would be different across languages. maybe it should be in a different section?

**Review by npentrel — APPROVED** [review]
Lgtm once Cheuk approves

File: docs/build/program/apis/robot.md
Comment (sguequierre) [reply docs/build/program/apis/robot.md]: @cheukt got it thank you!

File: docs/build/program/apis/robot.md
Comment (sguequierre) [inline docs/build/program/apis/robot.md:263]:
```suggestion

async def connect_with_channel() -> RobotClient:
```

File: docs/build/program/apis/robot.md
Comment (sguequierre) [inline docs/build/program/apis/robot.md:196]:
```suggestion
api_key_id  = '<API-KEY-ID>'
```

File: docs/build/program/apis/robot.md
Comment (sguequierre) [inline docs/build/program/apis/robot.md:193]:
```suggestion
api_key = '<API-KEY>'
```

**Pull request merged**
